### ISSUE TITLE
Sunshine doesn't seem to detect NvFBC (Arch Linux + nvidia-patch)

### ISSUE THREAD

**Issue by pharmville** [LizardByte/Sunshine#2205]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

See screenshot below. Left side shows that sunshine works, because I allow autodetect, which chooses X11.

Right side top, you can see I tested NvFBC with one of Nvidia's sample apps, and it works (it creates bitmap file of each frame it captures).

Right side lower, you can see I try to force NvFBC in Sunshine settings, and it doesn't work.

![Screenshot from 2024-03-03 17-33-31](https://github.com/LizardByte/Sunshine/assets/2756682/341e86be-3ac3-485e-8501-2b1dca665a6c)

### Expected Behavior

Sunshine should use NvFBC when available, not X11.

### Additional Context

I haven't tested on any other operating systems or hardware. Although I did try the flatpak version at one point.

### Host Operating System

Linux

### Operating System Version

Arch Linux (latest 6.7 kernel)

### Architecture

64 bit

### Sunshine commit or version

0.21 (nightly didn't work either)

### Package

Linux - pkg.tar.zst

### GPU Type

Nvidia

### GPU Model

GeForce RTX 3050

### GPU Driver/Mesa Version

NVIDIA Driver Version: 550.54.14, CUDA Version: 12.4

### Capture Method (Linux Only)

NvFBC

### Config

```Shell
av1_mode = 1
fps = [30,60]
resolutions = [
    1280x720,
    1920x1080,
    3840x2160
]
capture = nvfbc
```

### Relevant log output

```shell
[capture] -- [nvfbc]
[resolutions] -- [[
    1280x720,
    1920x1080,
    3840x2160
]]
[fps] -- [[30,60]]
[av1_mode] -- [1]
[2024:03:03:19:15:03]: Info: Sunshine version: 0.21.0
[2024:03:03:19:15:03]: Error: Unable to initialize capture method
[2024:03:03:19:15:03]: Error: Platform failed to initialize
[2024:03:03:19:15:03]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024:03:03:19:15:03]: Info: Trying encoder [nvenc]
[2024:03:03:19:15:03]: Info: System tray created
[2024:03:03:19:15:04]: Info: Encoder [nvenc] failed
[2024:03:03:19:15:04]: Info: Trying encoder [vaapi]
[2024:03:03:19:15:04]: Info: Encoder [vaapi] failed
[2024:03:03:19:15:04]: Info: Trying encoder [software]
[2024:03:03:19:15:05]: Info: Encoder [software] failed
[2024:03:03:19:15:05]: Fatal: Couldn't find any working encoder
[2024:03:03:19:15:05]: Error: Video failed to find working encoder
[2024:03:03:19:15:05]: Info: Adding avahi service Sunshine
[2024:03:03:19:15:05]: Info: Configuration UI available at [https://localhost:47990]
[2024:03:03:19:15:06]: Info: Avahi service Sunshine successfully established.
[2024:03:03:19:15:36]: Info: Opening UI from system tray
[2024:03:03:19:15:36]: Info: Opened url [https://localhost:47990]
[2024:03:03:19:16:41]: Info: Quitting from system tray
[2024:03:03:19:16:41]: Info: Interrupt handler called
```

**Comment by cgutman** [comment]
I don't see any log messages from the CUDA code, so you probably didn't build with the required dependencies for the CUDA code to be enabled.

Please attach your CMake output.

**Comment by ReenigneArcher** [comment]
If using the pre-built pkg.tar.zst it should have been compiled with cuda. https://github.com/LizardByte/Sunshine/actions/runs/8134829149/job/22228380077#step:10:1020

Looks like environment variable `WAYLAND_DISPLAY` is not defined... is this a headless or SSH session?

**Comment by pharmville** [comment]
I took the advice to compile it myself, and it works. Thank you! Although I feel like I wasted your time, I appreciate your help. I didn't even see the prebuilt pkg.tar.zst... that probably would have saved me some time. Looking into what package I used - now I see it wasn't from the official arch repositories; I forgot about the cachyos repos I added a few days back. Therefore it seems the cachyos and flathub versions potentially have issues, but cloning the git repo and installing from source - everything works. Thanks again.
